abci/example/kvstore/kvstore.go

```diff
@@ -117,6 +121,11 @@ func (app *Application) DeliverTx(req types.RequestDeliverTx) types.ResponseDeli
 }
 
 func (app *Application) CheckTx(req types.RequestCheckTx) types.ResponseCheckTx {
+	if req.Type == types.CheckTxType_Recheck {
```

Comment:
Are we sure we want this if?
It would also make sense if we got a replay attempt for a transaction, wouldn't it?

Reply:
We clear the list of transactions to remove after every height, so this really only pertains to the recheck attempts that occur directly after the height. Replay attempts may happen but at the moment, the application does guard against those at all, so I don't think we need to add that extra functionality to this code.